components/shared/embedder/lib.rs

```diff
@@ -585,3 +589,57 @@ pub enum LoadStatus {
     /// See <https://developer.mozilla.org/en-US/docs/Web/API/Document/readyState>
     Complete,
 }
+
+/// <https://notifications.spec.whatwg.org/#notifications>
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct Notification {
+    /// <https://notifications.spec.whatwg.org/#concept-title>
+    pub title: String,
+    /// <https://notifications.spec.whatwg.org/#body>
+    pub body: String,
+    /// <https://notifications.spec.whatwg.org/#tag>
+    pub tag: String,
+    /// <https://notifications.spec.whatwg.org/#concept-direction>
+    pub dir: NotificationDirection,
+    /// <https://notifications.spec.whatwg.org/#concept-language>
+    pub lang: String,
+    /// <https://notifications.spec.whatwg.org/#require-interaction-preference-flag>
+    pub require_interaction: bool,
+    /// <https://notifications.spec.whatwg.org/#silent-preference-flag>
+    pub silent: Option<bool>,
+    /// <https://notifications.spec.whatwg.org/#icon-url>
+    pub icon_url: Option<ServoUrl>,
+    /// <https://notifications.spec.whatwg.org/#icon-resource>
+    pub icon_resource: Option<Arc<Image>>,
+    /// <https://notifications.spec.whatwg.org/#badge-url>
+    pub badge_url: Option<ServoUrl>,
+    /// <https://notifications.spec.whatwg.org/#badge-resource>
+    pub badge_resource: Option<Arc<Image>>,
+    /// <https://notifications.spec.whatwg.org/#image-url>
+    pub image_url: Option<ServoUrl>,
+    /// <https://notifications.spec.whatwg.org/#image-resource>
+    pub image_resource: Option<Arc<Image>>,
+    /// <https://notifications.spec.whatwg.org/#actions>
+    pub actions: Vec<NotificationAction>,
+}
+
+/// <https://notifications.spec.whatwg.org/#actions>
```

Comment:
The documentation for this should be better since this is exposed to the embedding API.